src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Windows.cs

```diff
@@ -73,7 +73,7 @@ private bool TryConnect(int timeout, CancellationToken cancellationToken)
                     }
 
                     // The timeout has expired.
-                    if (errorCode == Interop.Errors.ERROR_SUCCESS)
+                    if (errorCode == Interop.Errors.ERROR_SEM_TIMEOUT)
```

Comment:
Is there a test to validate this?

Reply:
Also, the `cancellationToken.CanBeCanceled` below looks wrong.  I expect that should be `cancellationToken.IsCancellationRequested`.

Reply:
TL;DR: Remove this business of `CanBeCanceled` and coalesce `errorCode == Interop.Errors.ERROR_SEM_TIMEOUT` with the one above to return false.

Regarding the `CanBeCanceled` usage: it is a convoluted logic but this method is called from a loop that if used a `cancellationToken` that can be cancelled and a large enough timeout (>50ms), it slices the timeout in small chuncks so it can check for the a cancellation being raised from the token when the sliced timeout happens. 

The important thing is that in a timeout of the shorter slice it returns false but the loop calling the method already has logic to handle and throw `TimeoutException`...  anyway since it was checking for success instead of timeout that code path was never being triggered.